File: Swift-C++-IR-Comparison/test/arithmetic-operators/Addition/Swift/OverflowAdd.swift
@main
struct MyMain {

  static func main() {
    // This call would crash with a normal add.
    // returns -9223372036854775808
    let a = addWithOverflow(x: Int.max, y: 1)
  }

  // The & sign turns an addition to an addition without overflow
  static func addWithOverflow(x: Int, y: Int) -> Int {
    let tmp = x &+ y
    return tmp
  }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-1/OverflowAdd.cpp


#include <iostream>

struct MyMain {
    static void main() {
        // This call would crash with a normal add.
        // returns -9223372036854775808
        int result = addWithOverflow(std::numeric_limits<int>::max(), 1);
    }

    // The & sign turns an addition to an addition without overflow
    static int addWithOverflow(int x, int y) {
        int tmp = x &+ y;
        return tmp;
    }
};

int main() {
    MyMain::main();
    return 0;
}